beetmoverscript/script.py

```diff
@@ -61,33 +79,41 @@
     for locale in artifacts_to_beetmove:
         for artifact in artifacts_to_beetmove[locale]:
             source = artifacts_to_beetmove[locale][artifact]
+            pretty_name = manifest['mapping'][locale][artifact]['s3_key']
             dest_dated = os.path.join(manifest["s3_prefix_dated"],
-                                      manifest['mapping'][locale][artifact]['s3_key'])
+                                      pretty_name)
             dest_latest = os.path.join(manifest["s3_prefix_latest"],
-                                       manifest['mapping'][locale][artifact]['s3_key'])
+                                       pretty_name)
             balrog_manifest = manifest['mapping'][locale][artifact].get('update_balrog_manifest')
             beets.append(
                 asyncio.ensure_future(
                     move_beet(context, source, destinations=(dest_dated, dest_latest),
-                              locale=locale, update_balrog_manifest=balrog_manifest)
+                              locale=locale, update_balrog_manifest=balrog_manifest,
+                              pretty_name=pretty_name)
                 )
             )
     await raise_future_exceptions(beets)
 
 
-async def move_beet(context, source, destinations, locale, update_balrog_manifest):
-    beet_config = deepcopy(context.config)
-    beet_config.setdefault('valid_artifact_task_ids', context.task['dependencies'])
-
+async def move_beet(context, source, destinations, locale,
+                    update_balrog_manifest, pretty_name):
     await retry_upload(context=context, destinations=destinations, path=source)
 
+    context.checksums_dict.setdefault(pretty_name, {
```

Comment:
I think you're using `setdefault` to avoid setting the checksum for a file multiple times.  If you didn't care about that, you could have

```python
context.checksums_dict[pretty_name] = {algo: get_hash(source, algo) for algo in CHECKSUMS_DIGESTS}
```

instead.

However, the right side of `setdefault()` gets evaluated either way.  For instance,

```python
#!/usr/bin/env python
from __future__ import print_function
d = {'a': 1}
d.setdefault('a', print("foo"))  # This shouldn't set d['a'], because it's already set
print(d)
```

The output from the above scriptlet is

```
foo
{'a': 1}
```

So the dictionary `d` isn't modified because we used `setdefault`, and `d['a']` is already set.  However, we evaluate the `print(foo)` anyway.  That's pretty harmless here, but this means is we'll run `get_hash` for all the `CHECKSUMS_DIGESTS`, even if `contest.checksums_dict[pretty_name]` is already defined, wasting cycles.

Instead, (if we are worried about running `get_hash` for the same file multiple times), we could

```python
if context.checksums_dict.get(pretty_name) is None:
    context.checksums_dict[pretty_name] = {algo: get_hash(source, algo) for algo in CHECKSUMS_DIGESTS}
```

I think that'll do what you want.